nilearn/load_confounds/tests/test_parser.py

```diff
@@ -0,0 +1,476 @@
+    img, mask_conf, mask_rand, X = _simu_img(demean=True)
+
+    # Check that most variance is removed
+    # in voxels composed of pure confounds
+    tseries_std = _tseries_std(img, mask_conf, X, None, False)
+    assert np.mean(tseries_std < 0.0001)
+
+    # Check that most variance is preserved
+    # in voxels composed of random noise
+    tseries_std = _tseries_std(img, mask_rand, X, None, False)
+    assert np.mean(tseries_std > 0.9)
+
+
+@pytest.mark.filterwarnings("ignore")
+def test_nilearn_standardize_zscore():
+    """Test removing confounds in nilearn with zscore standardization."""
+    # Simulate data
+
+    img, mask_conf, mask_rand, X = _simu_img(demean=True)
+
+    # We now load the time series with vs without confounds
+    # in voxels composed of pure confounds
+    # the correlation before and after denoising should be very low
+    # as most of the variance is removed by denoising
+    tseries_raw, tseries_clean = _denoise(img, mask_conf, X, None, "zscore")
+    corr = _corr_tseries(tseries_raw, tseries_clean)
+    assert corr.mean() < 0.2
+
+    # We now load the time series with zscore standardization
+    # with vs without confounds in voxels where the signal is uncorrelated
+    # with confounds. The correlation before and after denoising should be very
+    # high as very little of the variance is removed by denoising
+    tseries_raw, tseries_clean = _denoise(img, mask_rand, X, None, "zscore")
+    corr = _corr_tseries(tseries_raw, tseries_clean)
+    assert corr.mean() > 0.8
+
+
+def test_nilearn_standardize_psc():
+    """Test removing confounds in nilearn with psc standardization."""
+    # Similar test to test_nilearn_standardize_zscore, but with psc
+    # Simulate data
+
+    img, mask_conf, mask_rand, X = _simu_img(demean=False)
+
+    # Areas with confound
+    tseries_raw, tseries_clean = _denoise(img, mask_conf, X, None, "psc")
+    corr = _corr_tseries(tseries_raw, tseries_clean)
+    assert corr.mean() < 0.2
+
+    # Areas with random noise
+    tseries_raw, tseries_clean = _denoise(img, mask_rand, X, None, "psc")
+    corr = _corr_tseries(tseries_raw, tseries_clean)
+    assert corr.mean() > 0.8
+
+
+def test_confounds2df():
+    """Check auto-detect of confonds from an fMRI nii image."""
+    conf = lc.Confounds()
+    file_confounds_nii = os.path.join(
+        path_data, "test_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz"
+    )
+    conf.load(file_confounds_nii)
+    assert "trans_x" in conf.confounds_.columns
+
+
+def test_sanitize_strategy():
+    """Check that flawed strategy options generate meaningful error
+    messages."""
+    with pytest.raises(ValueError):
+        lc.Confounds(strategy="string")
+
+    with pytest.raises(ValueError):
+        lc.Confounds(strategy=["error"])
+
+    with pytest.raises(ValueError):
+        lc.Confounds(strategy=[0])
+
+    conf = lc.Confounds(strategy=["motion"])
+    assert "non_steady_state" in conf.strategy
+
+
+def test_motion():
```

Comment:
I'm wondering whether we should unit test low level helper functions (at least some of them...). WDYT?

Reply:
Some of the `sample_mask` related functions would be really helpful with some extra tests I feel.